src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -1127,6 +1163,18 @@ private static ImmutableArray<(DiagnosticAnalyzer, ImmutableArray<SyntaxTreeAnal
             return builder.ToImmutableAndFree();
         }
 
+        private static ImmutableArray<(DiagnosticAnalyzer, ImmutableArray<AdditionalFileAnalyzerAction>)> MakeAdditionalFileActionsByAnalyzer(in AnalyzerActions analyzerActions)
+        {
+            var builder = ArrayBuilder<(DiagnosticAnalyzer, ImmutableArray<AdditionalFileAnalyzerAction>)>.GetInstance();
+            var actionsByAnalyzers = analyzerActions.AdditionalFileActions.GroupBy(action => action.Analyzer);
+            foreach (var analyzerAndActions in actionsByAnalyzers)
+            {
+                builder.Add((analyzerAndActions.Key, analyzerAndActions.ToImmutableArray()));
+            }
+
+            return builder.ToImmutableAndFree();
```

Comment:
This pattern appears in multiple `Make*ByAnalyzer()` methods. Would it make sense to extract a helper method?

Reply:
Thanks, that simplified quite a bit.